scenes/game_elements/props/door/components/door.gd

```diff
@@ -0,0 +1,23 @@
+# SPDX-FileCopyrightText: The Threadbare Authors
+# SPDX-License-Identifier: MPL-2.0
+@tool
+extends Toggleable
+
+@export var opened: bool = false:
+	set(new_val):
+		opened = new_val
+		update_opened_state()
+
+
+func set_toggled(value: bool) -> void:
+	opened = value
+
+
+func update_opened_state() -> void:
+	%DoorClosed.visible = !opened
+	%DoorOpened.visible = opened
+
+	%ColliderWhenOpened.set_collision_layer_value(5, opened)
+	%ColliderWhenOpened.set_collision_mask_value(1, opened)
+	%ColliderWhenClosed.set_collision_layer_value(5, !opened)
+	%ColliderWhenClosed.set_collision_mask_value(1, !opened)
```

Comment:
I think I'd prefer it if the frame of the door didn't detect collisions when the door is open. It makes sense for it to collide but it feels a bit clunky in my opinion.